### PR TITLE
[Feature] support ada module and training

### PR DESCRIPTION
**Pull request by plyfager**
1. Support `ada` augment.
2. Support `ADAStyleGAN2Discriminator` and its training.

### CODE REVIEW

File: mmgen/models/gans/static_unconditional_gan.py [hunk 1]
```diff
@@ -270,6 +270,17 @@ def train_step(self,
         else:
             optimizer['generator'].step()
 
+        # update ada p
+        if hasattr(self.discriminator,
```
Comment (LeoXing1996) [inline mmgen/models/gans/static_unconditional_gan.py:274]: Those lines can be covered in `test_static_unconditional_gan.py`.

File: mmgen/models/architectures/stylegan/modules/styleganv3_modules.py [hunk 2]
```diff
@@ -517,6 +523,18 @@ def forward(self,
 
     @staticmethod
     def design_lowpass_filter(numtaps, cutoff, width, fs, radial=False):
+        """Design lowpass filter giving related arguments.,
+        Args:
```
Comment (LeoXing1996) [inline mmgen/models/architectures/stylegan/modules/styleganv3_modules.py:527]: Add a blank line here.

File: mmgen/models/architectures/stylegan/modules/styleganv3_modules.py [hunk 3]
```diff
@@ -517,6 +523,18 @@ def forward(self,
...
+            numtaps (int): Length of the filter. `numtaps` must be odd if a
+                passband includes the Nyquist frequency.
+            cutoff (float): Cutoff frequency of filter
+            width (float): The approximate width of the transition region.
+            fs (float): The sampling frequency of the signal.
+            radial (bool, optional):  Whether use radially symmetric jinc-based
+                filter. Defaults to False.
+        Returns:
```
Comment (LeoXing1996) [inline mmgen/models/architectures/stylegan/modules/styleganv3_modules.py:535]: Please add a blank line before `Returns`.

File: mmgen/models/architectures/stylegan/modules/styleganv3_modules.py [hunk 4]
```diff
@@ -461,13 +461,19 @@ def __init__(
...
-    def forward(self,
-                x,
-                w,
-                noise_mode='random',
-                force_fp32=False,
-                update_emas=False):
-        assert noise_mode in ['random', 'const', 'none']  # unused
+    def forward(self, x, w, force_fp32=False, update_emas=False):
+        """Forward function for synthesis layer.
+        Args:
```
Comment (LeoXing1996) [inline mmgen/models/architectures/stylegan/modules/styleganv3_modules.py:466]: Blank line here.

File: mmgen/core/evaluation/metrics.py [hunk 5]
```diff
@@ -1278,6 +1285,16 @@ def summary(self):
         return ppl_score
 
     def get_sampler(self, model, batch_size, sample_model):
+        """Get sampler for sampling along the path.
+
+        Args:
+            model (nn.Module): Generative model.
+            batch_size (int): Sampling batch size.
+            sample_model (str): Which model you want to use. ['ema',
+            'orig']. Defaults to 'ema'.
```
Comment (LeoXing1996) [inline mmgen/core/evaluation/metrics.py:1294]: We need to indent here.

File: configs/styleganv3/stylegan3_t_noaug_fp16_gamma32.8_ffhq_1024_b4x8.py [hunk 6]
```diff
@@ -56,7 +56,7 @@
         inception_args=dict(type='StyleGAN'),
         bgr2rgb=True))
 
-inception_path = '/mnt/lustre/yangyifei1/repos/fix-s3/work_dirs/cache/inception-2015-12-05.pt'  # noqa
+inception_path = None  # set by user
```
[the remark this answers is not in the document]
Comment (plyfager) [reply configs/styleganv3/stylegan3_t_noaug_fp16_gamma32.8_ffhq_1024_b4x8.py:59]: True. This is used for a no-net environment.

**Pull request merged**